lib/Conversion/PolyToStandard/PolyToStandard.cpp

```diff
@@ -113,6 +113,37 @@ struct ConvertToTensor : public OpConversionPattern<ToTensorOp> {
   }
 };
 
+struct ConvertConstant : public OpConversionPattern<ConstantOp> {
+  ConvertConstant(mlir::MLIRContext *context)
+      : OpConversionPattern<ConstantOp>(context) {}
+
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      ConstantOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    RankedTensorType tensorType = cast<RankedTensorType>(
+        typeConverter->convertType(op.getResult().getType()));
+    PolynomialAttr attr = op.getInput();
+    SmallVector<Attribute> coeffs;
+    auto eltTy = tensorType.getElementType();
+    unsigned numTerms = tensorType.getShape()[0];
+    coeffs.reserve(numTerms);
+    // This is inefficient for large-degree polys, but as of this writing we
+    // don't have a lowering that uses a sparse representation.
+    for (int i = 0; i < numTerms; ++i) {
+      coeffs.push_back(rewriter.getIntegerAttr(eltTy, 0));
+    }
+    for (const auto &term : attr.getPolynomial().getTerms()) {
+      coeffs[term.exponent.getZExtValue()] = rewriter.getIntegerAttr(
+          eltTy, term.coefficient.trunc(eltTy.getIntOrFloatBitWidth()));
```

Comment:
Does this need a verifier that the coefficient terms have size less than the modulus bit width? or else we can APInt::mod them here? (also, are they potentially negative? test case just uses small positive int)

Reply:
Yes, I think I should add a verifier for poly.constant to make sure the input polynomial agrees with the ring in all senses (low enough degree, coeffs, etc.) adding this as issue https://github.com/google/heir/issues/190